### PR TITLE
Platform 935 analytics upgrade

### PR DESCRIPTION
**Pull request by harnash**
Separating dev+staging and production environments when collecting Google Analytics stats. Before all stats were sent to production accounts (on Google Analytics side. There was "almost" identical piece of code for collecting stats for development environments but it was never used. This PR cleans up the code, merges those two files and properly directs stats to production and development accounts accordingly.

This is the first part of the migration process to Universal Analytics.

### CODE REVIEW

**Comment by harnash** [comment]
@macbre: PR is ready for review

File: extensions/wikia/AnalyticsEngine/AnalyticsProviderGAS.php [hunk 1]
```diff
@@ -1,6 +1,28 @@
 <?php
+
+global $wgHooks;
+
+$wgHooks['MakeGlobalVariablesScript'][] = 'AnalyticsProviderGAS::wfMakeGlobalVariablesScript';
```
Comment (macbre) [inline extensions/wikia/AnalyticsEngine/AnalyticsProviderGAS.php]: Please move hook registration to `/extensions/wikia/AnalyticsEngine/AnalyticsEngine.setup.php`

File: extensions/wikia/AnalyticsEngine/AnalyticsProviderGAS.php [hunk 2]
```diff
@@ -1,6 +1,28 @@
...
+
 class AnalyticsProviderGAS implements iAnalyticsProvider {
 
+	/**
+	 * Hook for setting proper Google Analytics JS variables.
+	 *
+	 * @param array $vars JS variables to be added at the bottom of the page
+	 * @return bool return true - it's a hook
+	 */
+	public static function wfMakeGlobalVariablesScript(Array &$vars) {
```
Comment (macbre) [inline extensions/wikia/AnalyticsEngine/AnalyticsProviderGAS.php]: We use `on<hook name>` naming convention.

File: extensions/wikia/AnalyticsEngine/AnalyticsProviderGAS.php [hunk 3]
```diff
@@ -1,6 +1,28 @@
...
+		global $wgDevelEnvironment, $wgStagingEnvironment;
+
+		// Enable collecting stats to staging accounts on all dev and staging environments
+		if ($wgDevelEnvironment || $wgStagingEnvironment) {
+			$vars['wgGaStaging'] = true;
```
Comment (macbre) [inline extensions/wikia/AnalyticsEngine/AnalyticsProviderGAS.php]: :+1:

File: extensions/wikia/AnalyticsEngine/AnalyticsProviderGAS.php (hunk 1 above)
Comment (harnash) [inline extensions/wikia/AnalyticsEngine/AnalyticsProviderGAS.php]: done in a73759861e2a7414f409a61a85ae056525948a74

File: extensions/wikia/AnalyticsEngine/AnalyticsProviderGAS.php (hunk 2 above)
Comment (harnash) [inline extensions/wikia/AnalyticsEngine/AnalyticsProviderGAS.php]: done in 28968e6f6abc04bf99eca6846b7b153bc02536f4

**Comment by macbre** [comment]
LGTM

**Comment by harnash** [comment]
@macbre: Just small fix. It turns out that MakeGlobalVariablesScript hook is a bit late for declaring global variables in GAS script.

**Comment by macbre** [comment]
Good catch!

**Pull request merged**